moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -128,6 +128,7 @@ MotionPlanningFrame::MotionPlanningFrame(MotionPlanningDisplay* pdisplay, rviz::
   connect(ui_->planning_scene_tree, SIGNAL(itemChanged(QTreeWidgetItem*, int)), this,
           SLOT(warehouseItemNameChanged(QTreeWidgetItem*, int)));
   connect(ui_->reset_db_button, SIGNAL(clicked()), this, SLOT(resetDbButtonClicked()));
+  connect(ui_->add_shape_button, &QPushButton::clicked, [this]() { addPrimitiveShape(); });
```

Comment:
```suggestion
  connect(ui_->add_shape_button, &QPushButton::clicked, this, &MotionPlanningFrame::addPrimitiveShape);
```

No need for a lambda here. Also please be careful with this short version as `this` or other pointers bound in the lambda could be destroyed elsewhere and then crash on calling the lambda. There is a preferred variant with four arguments where the third denotes the scope of the connection and if either sender or receiver are destructed:
```
connect(ui_->add_shape_button, &QPushButton::clicked, this, [this]() { addPrimitiveShape(); });
```
But as stated above, this shouldn't be necessary here

Reply:
@simonschmeisser  The `this` pointer points to the panel itself so I don't see how it could ever get destroyed in this context but I'll make the change

Reply:
You're right, in this context it won't happen. `ui_->shape_button` is a child of this and therefore gets destroyed before `this` and cancels the `connection` ... just wanted to point out the general danger of that variant of `connect()`

My main remark was about the number of brackets and the unnecessary lambda however :)